libsolidity/analysis/TypeChecker.cpp

```diff
@@ -4024,18 +4041,43 @@ void TypeChecker::endVisit(Literal const& _literal)
 					// This was already validated in visit(Literal) but the error is not fatal.
 					if (!mantissa || !exponent)
 						solAssert(!m_errorReporter.errors().empty());
-					else if (
-						!mantissa->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(0)) ||
-						!exponent->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(1))
-					)
-						// TODO: Is this triggered when the argument is out of range? Test.
-						parameterTypeMessage = "The type of the literal cannot be converted to the parameters of the suffix function.";
+					// TODO: Is this triggered when the argument is out of range? Test.
+					else if (!mantissa->isImplicitlyConvertibleTo(*suffixFunctionType->parameterTypes().at(0)))
+						parameterTypeMessage = fmt::format(
+							"Type {} of the mantissa cannot be converted to type {} of the suffix function parameter.",
```

Comment:
I'd make this closer to the other message, because this is really the same underlying issue:

```suggestion
					"This fractional number cannot be decomposed into a mantissa and decimal exponent "
					"that fit the range of parameters of the suffix function. "
					"The mantissa is out of range of type {}.",
```
Same for exponent below.

Reply:
Right. I was actually in doubt if I should use the mobile type. So it's clear to me now.